pkg/storaged/utils.js

```diff
@@ -757,6 +884,37 @@ export function get_active_usage(client, path, top_action, child_action) {
     return usage;
 }
 
+async function set_fsys_noauto(client, block, mount_point) {
+    for (const conf of block.Configuration) {
+        if (conf[0] == "fstab" &&
+            decode_filename(conf[1].dir.v) == mount_point) {
+            const options = parse_options(get_block_mntopts(conf[1]));
+            if (options.indexOf("noauto") >= 0)
+                continue;
+            options.push("noauto");
+            const new_conf = [
+                "fstab",
+                Object.assign({ }, conf[1],
+                              {
+                                  opts: {
+                                      t: 'ay',
+                                      v: encode_filename(unparse_options(options))
```

Comment:
These 3 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19388-20230928-071616-0731117d-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)